src/components/erc721/collectibles/collectible_list_modal.tsx

```diff
@@ -0,0 +1,178 @@
+    border-color: ${props => props.theme.componentsTheme.marketsSearchFieldBackgroundColor};
+    color: ${props => props.theme.componentsTheme.marketsSearchFieldTextColor};
+    max-width: 100%;
+    width: ${modalWidth};
+
+    ::placeholder {
+        text-align: left;
+    }
+`;
+
+const CloseModalButtonStyle = styled(CloseModalButton)`
+    margin: 0;
+`;
+
+const initialState = {
+    currentCollectible: null,
+    filterText: '',
+};
+
+// PureComponents only rerender if at least one state or prop value changes.
+class CollectibleListModalContainer extends React.PureComponent<Props, State> {
+    public state: State = {
+        ...initialState,
+    };
+
+    // The render method on this PureComponent is called only if props or state.filterText has changed.
+    public render = () => {
+        const { theme, userCollectibles, isCollectibleModalOpen } = this.props;
+        const { filterText } = this.state;
+        const collectibles = Object.keys(userCollectibles).map(key => userCollectibles[key]);
+        const filteredCollectibles = this._filterCollectiblesByName(collectibles, filterText);
+
+        let content = null;
+        if (filteredCollectibles) {
+            content = filteredCollectibles.map((item, index) => {
+                return (
+                    <CollectibleOnListContainer
+                        collectible={item}
+                        isListItem={true}
+                        key={index}
+                        onClick={this._closeModal}
+                    />
+                );
+            });
+        }
+
+        return (
+            <Modal isOpen={isCollectibleModalOpen} style={theme.modalTheme} onRequestClose={this._closeModal}>
+                <ModalTitleWrapper>
+                    <ModalTitleTop>
+                        <ModalTitle>Selling Item</ModalTitle>
+                        <CloseModalButtonStyle onClick={this._closeModal} />
+                    </ModalTitleTop>
+                    <SearchStyled placeHolder={'Search Wallet'} onChange={this._handleSearchInputChanged} />
+                </ModalTitleWrapper>
+                <ModalContent>{content ? content : <CardLoading minHeight={modalContentHeight} />}</ModalContent>
+            </Modal>
+        );
+    };
+
+    private readonly _closeModal = () => {
+        this.props.toggleCollectibleListModal();
+    };
+
+    private readonly _handleSearchInputChanged = (event: any) => {
+        const newInputValue = event && event.target ? event.target.value : '';
+
+        this.setState({
+            filterText: newInputValue,
+        });
+    };
+
+    private readonly _filterCollectiblesByName = (collectibles: Collectible[], name: string): Collectible[] => {
+        return collectibles.filter((collectible: Collectible) => this._filterCollectibleByNameFn(collectible, name));
+    };
+
+    private readonly _filterCollectibleByNameFn = (collectible: Collectible, name: string): boolean => {
```

Comment:
This could be exported on an utility function and used like [this PR](https://github.com/0xProject/0x-launch-kit-frontend/pull/418), thoughts @unjapones ?

Reply:
It's merged now